src/setup/rgw.py

```diff
@@ -122,14 +124,31 @@ def config(conf: MappedConf):
             Rgw._update_rgw_config_with_endpoints(conf, service_endpoints, instance)
             instance = instance + 1
 
-        Log.info('Config phase completed.')
-        return 0
+        # Read Motr HA endpoint from data pod using hctl fetch-fids and update in config file
+        Log.info('Reading motr_ha_endpoint from data pod')
+        config_path = Rgw._get_cortx_conf(conf, CONFIG_PATH_KEY)
+        hare_config_dir = os.path.join(config_path, 'hare', 'config', Rgw._machine_id)
+        data_pod_hostname = socket.gethostname().replace('server', 'data')
+
+        fetch_fids_cmd = f'hctl fetch-fids -c {hare_config_dir} --node {data_pod_hostname}'
+        out, err, rc = SimpleProcess(fetch_fids_cmd).run()
+        if rc != 0:
+            Log.error(f'Unable to read fid information for hostname: '
+                f'{data_pod_hostname}. {err}')
+            raise SetupError(rc, 'Unable to read fid information for hostname: '
+                '%s. %s', data_pod_hostname, err)
+        decoded_out = json.loads(out.decode('utf-8'))
+        motr_ha_endpoint = [endpoints['ep'] for endpoints in decoded_out \
+            if 'hax' in endpoints.values()][0]
+        Log.info(f'Fetched motr_ha_endpoint from data pod. Endpoint: {motr_ha_endpoint}')
 
-    @staticmethod
-    def start(conf: MappedConf, index: str):
-        """Create rgw admin user and start rgw service."""
+        rgw_config_path = Rgw._get_rgw_config_path(conf)
+        Rgw._load_rgw_config(Rgw._rgw_conf_idx, f'ini://{rgw_config_path}')
+        Conf.set(Rgw._rgw_conf_idx, \
+            f'client.radosgw-admin>{RgwEndpoint.MOTR_HA_EP.value}', motr_ha_endpoint)
+        Conf.save(Rgw._rgw_conf_idx)
 
-        Log.info('Create rgw admin user and start rgw service.')
+        Log.info(f'Updated motr_ha_endpoint in config file {rgw_config_path}')
         # admin user should be created only on one node.
```

Comment:
Should we move user creation in 'init' phase ?

Reply:
@cdeshmukh Please confirm